rust/worker/src/execution/operators/brute_force_knn.rs

```diff
@@ -134,7 +139,11 @@ impl Operator<BruteForceKnnOperatorInput, BruteForceKnnOperatorOutput> for Brute
             }
         };
         let log_materializer = LogMaterializer::new(record_segment_reader, input.log.clone(), None);
-        let logs = match log_materializer.materialize().await {
+        let logs = match log_materializer
+            .materialize()
+            .instrument(tracing::info_span!("Materialize logs"))
```

Comment:
will it be useful to have the log size as an atrribute here?

Reply:
Also, the operation that is needing materialization?

Reply:
> will it be useful to have the log size as an atrribute here? sure I can add that inside of the log_materializer itself - I think that should be an event inside the materializer

>  the operation that is needing materialization?
This should already be in the enclosing trace? I am a bit unsure about our conventions here. But looking at the trace it should be easy to tell which operation is being used. I'm inclined to keep it without that as part of the name, since then we can group by on this